File: frontend/src/services/adminService.js
import axiosClient from '../api/axiosClient'

export function getAllStudents() {
  return axiosClient.get('/admin/students')
}

export function addMilestoneToStudent(studentId, payload) {
  return axiosClient.post(`/admin/students/${studentId}/milestones`, payload)
}

export function updateMilestone(milestoneId, payload) {
  return axiosClient.put(`/admin/milestones/${milestoneId}`, payload)
}

export function deleteMilestone(milestoneId) {
  return axiosClient.delete(`/admin/milestones/${milestoneId}`)
}

export function giveFeedbackToStudent(studentId, payload) {
  return axiosClient.post(`/admin/students/${studentId}/feedbacks`, payload)
}

export function getAllFeedbacks() {
  return axiosClient.get('/admin/feedbacks')
}

export function approveProject(projectId) {
  return axiosClient.put(`/admin/projects/approve/${projectId}`)
}

File: backend/src/main/java/com/fsad/portal/service/AdminService.java
package com.fsad.portal.service;

import com.fsad.portal.dto.FeedbackDto;
import com.fsad.portal.dto.FeedbackRequestDto;
import com.fsad.portal.dto.MilestoneDto;
import com.fsad.portal.dto.MilestoneRequestDto;
import com.fsad.portal.dto.ProjectDto;
import com.fsad.portal.dto.UserDto;

import java.util.List;

public interface AdminService {
    List<UserDto> getAllStudents();
    MilestoneDto addMilestoneToStudent(Long studentId, MilestoneRequestDto request);
    MilestoneDto updateMilestone(Long milestoneId, MilestoneRequestDto request);
    void deleteMilestone(Long milestoneId);
    FeedbackDto giveFeedbackToStudent(Long studentId, FeedbackRequestDto request);
    List<FeedbackDto> getAllFeedbacks();
    ProjectDto approveProject(Long projectId);
}
